### PR TITLE
Improving the accessibility for lift instructions

### PR DESCRIPTION
**Pull request by seancurtis**

### CODE REVIEW

File: src/view/use-lift-instruction/use-lift-instruction.js [hunk 1]
```diff
@@ -24,8 +23,8 @@ export default function useLiftInstruction(
       // add the description text
       el.textContent = liftInstruction;
 
-      // hide the element visually
-      Object.assign(el.style, visuallyHidden);
```
Comment (alexreardon) [inline src/view/use-lift-instruction/use-lift-instruction.js]: Ah okay. So there are some things we want visually hidden, but available to the screen readers. And there are other things we want visually hidden + not available to screen readers

File: test/unit/integration/accessibility/axe-audit.spec.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+// @flow
```
Comment (alexreardon) [inline test/unit/integration/accessibility/axe-audit.spec.js]: 👏

File: src/view/use-lift-instruction/use-lift-instruction.js [hunk 3]
```diff
@@ -24,8 +23,8 @@ export default function useLiftInstruction(
...
+      // prevent AT from idly reading it. Fixes #1568
```
Comment (alexreardon) [inline src/view/use-lift-instruction/use-lift-instruction.js]:
```suggestion
// Visually hide the element and prevent screen readers from reading this element in the document flow
// This element is used as a `aria-labelledby` reference for *other elements* and will be read out for those
```

File: src/view/use-lift-instruction/use-lift-instruction.js [hunk 4]
```diff
@@ -24,8 +23,8 @@ export default function useLiftInstruction(
...
+      Object.assign(el.style, { display: 'none' });
```
Comment (alexreardon) [inline src/view/use-lift-instruction/use-lift-instruction.js]: Can you confirm that the lift instructions still work as expected?

File: src/view/use-lift-instruction/use-lift-instruction.js (hunk 1 above)
Comment (seancurtis) [reply src/view/use-lift-instruction/use-lift-instruction.js]: Yeah if we reference it only via something like `aria-describedby` then it doesn't need to be visible. The `visuallyHidden` is mainly for things like icon buttons (using `aria-label` is better), and I believe it's needed for live regions too.

File: src/view/use-lift-instruction/use-lift-instruction.js (hunk 4 above)
Comment (seancurtis) [reply src/view/use-lift-instruction/use-lift-instruction.js]: Works fine in VO (90% sure it also works in NVDA/JAWS too, but didn't test)

**Comment by alexreardon** [comment]
It looks like the flow violation is minor: it doesn't understand the jest extension. We can add a `//$FlowFixMe` for now. Then we can get this in :)

File: test/unit/integration/accessibility/axe-audit.spec.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+expect.extend(toHaveNoViolations);
+
+// Need to investigate this further as it doesn't seem to pick up the same errors
+// which are found by react-axe.
+it('should not fail an aXe audit', async () => {
+  render(<App />);
+
+  const results = await axe(document.body);
+
```
Comment (alexreardon) [inline test/unit/integration/accessibility/axe-audit.spec.js]:
```suggestion
// $FlowFixMe - flow doesn't know about hte custom validator
```

File: test/unit/integration/accessibility/axe-audit.spec.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+// $FlowFixMe - flow doesn't know about hte custom validator
```
Comment (alexreardon) [inline test/unit/integration/accessibility/axe-audit.spec.js]:
```suggestion
  // $FlowFixMe - flow doesn't know about hte custom validator
```

File: src/view/use-lift-instruction/use-lift-instruction.js [hunk 7]
```diff
@@ -24,8 +23,9 @@ export default function useLiftInstruction(
       // add the description text
       el.textContent = liftInstruction;
 
-      // hide the element visually
-      Object.assign(el.style, visuallyHidden);
+// Visually hide the element and prevent screen readers from reading this element in the document flow
+// This element is used as a `aria-labelledby` reference for *other elements* and will be read out for those
```
Comment (alexreardon) [inline src/view/use-lift-instruction/use-lift-instruction.js]:
```suggestion
  // Visually hide the element and prevent screen readers from reading this element in the document flow Object.assign(el.style, visuallyHidden);
  // This element is used as a `aria-labelledby` reference for *other elements* and will be read out for those
```

**Pull request merged**